### ISSUE TITLE
Role Currently causes a failed provision

### ISSUE THREAD

**Issue by skyeBreach** [nextcloud/user_oidc#910]
Hi,

So i noticed that i was getting a provisioning error when creating a new account, it would seem that if the role is set then it refuses to allow provisioning. I have tracked it down to what seems like a bad event handler in /var/www/nextcloud/apps/user_oidc/lib/Service/ProvisioningService.php on line 306

```
// Update role
		$event = new AttributeMappedEvent(ProviderService::SETTING_MAPPING_ROLE, $idTokenPayload, $role);
		$this->eventDispatcher->dispatchTyped($event);
		$this->logger->debug('Role mapping event dispatched');
		if ($event->hasValue() && $event->getValue() !== null && $event->getValue() !== '') {
			$account->setProperty('role', $role, $event->getValue(), '1', '');
		}
```

I believe the `$role` in the set property is in the wrong place, as all others have them swapped (and called scope).

For added clarity the raw error log is as follows
```
{"reqId":"NgYCNhmth0vURCzkzwJt","level":3,"time":"2024-07-30T13:42:27+00:00","remoteAddr":"2.122.53.191","user":"--","app":"index","method":"GET","url":"/apps/user_oidc/code?code=9daed48c496a4c0a92722c145a46d39f&state=DDHCWCKSET0S4XOL52RUIKUL2TOAGGSB","message":"Invalid scope","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:128.0) Gecko/20100101 Firefox/128.0","version":"29.0.4.1","exception":{"Exception":"InvalidArgumentException","Message":"Invalid scope","Code":0,"Trace":[{"file":"/var/www/nextcloud/lib/private/Accounts/AccountProperty.php","line":45,"function":"setScope","class":"OC\\Accounts\\AccountProperty","type":"->"},{"file":"/var/www/nextcloud/lib/private/Accounts/Account.php","line":53,"function":"__construct","class":"OC\\Accounts\\AccountProperty","type":"->"},{"file":"/var/www/nextcloud/apps/user_oidc/lib/Service/ProvisioningService.php","line":306,"function":"setProperty","class":"OC\\Accounts\\Account","type":"->"},{"file":"/var/www/nextcloud/apps/user_oidc/lib/Controller/LoginController.php","line":545,"function":"provisionUser","class":"OCA\\UserOIDC\\Service\\ProvisioningService","type":"->"},{"file":"/var/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php","line":232,"function":"code","class":"OCA\\UserOIDC\\Controller\\LoginController","type":"->"},{"file":"/var/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php","line":138,"function":"executeController","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->"},{"file":"/var/www/nextcloud/lib/private/AppFramework/App.php","line":184,"function":"dispatch","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->"},{"file":"/var/www/nextcloud/lib/private/Route/Router.php","line":338,"function":"main","class":"OC\\AppFramework\\App","type":"::"},{"file":"/var/www/nextcloud/lib/base.php","line":1050,"function":"match","class":"OC\\Route\\Router","type":"->"},{"file":"/var/www/nextcloud/index.php","line":49,"function":"handleRequest","class":"OC","type":"::"}],"File":"/var/www/nextcloud/lib/private/Accounts/AccountProperty.php","Line":81,"message":"Invalid scope","exception":[],"CustomMessage":"Invalid scope"},"id":"66a8edee7d595"}
```
and the panel error message is,
![image](https://github.com/user-attachments/assets/02827208-d46d-45fd-9d92-c70151a36251)

**Comment by julien-nc** [comment]
Thanks for reporting this. The fix in #930 should do it.

**Comment by leon1995** [comment]
Hi, I have version `6.1.2` installed and when provisioning with the role I still get an internal server error
```json
{
    "reqId": "8ezopG7J2tPSA6swc8um",
    "level": 3,
    "time": "2024-12-28T09:32:06+01:00",
    "remoteAddr": "<redacted>",
    "user": "--",
    "app": "index",
    "method": "GET",
    "url": "/apps/user_oidc/code?code=99e351ca29524ce2acffb040168c66ab&state=NAFPYW3A0TJC8B5OCPVRA27WS97XJS46",
    "message": "Invalid scope",
    "userAgent": "<redacted>",
    "version": "30.0.4.1",
    "exception": {
        "Exception": "InvalidArgumentException",
        "Message": "Invalid scope",
        "Code": 0,
        "Trace": [
            {
                "file": "/var/www/html/lib/private/Accounts/AccountProperty.php",
                "line": 29,
                "function": "setScope",
                "class": "OC\\Accounts\\AccountProperty",
                "type": "->",
                "args": [
                    "1"
                ]
            },
            {
                "file": "/var/www/html/lib/private/Accounts/Account.php",
                "line": 34,
                "function": "__construct",
                "class": "OC\\Accounts\\AccountProperty",
                "type": "->",
                "args": [
                    "role",
                    "<redacted>",
                    "1",
                    "",
                    ""
                ]
            },
            {
                "file": "/var/www/html/custom_apps/user_oidc/lib/Service/ProvisioningService.php",
                "line": 278,
                "function": "setProperty",
                "class": "OC\\Accounts\\Account",
                "type": "->",
                "args": [
                    "role",
                    "<redacted>",
                    "1",
                    ""
                ]
            },
            {
                "file": "/var/www/html/custom_apps/user_oidc/lib/Controller/LoginController.php",
                "line": 572,
                "function": "provisionUser",
                "class": "OCA\\UserOIDC\\Service\\ProvisioningService",
                "type": "->",
                "args": [
                    "<redacted>",
                    1,
                    {
                        "__class__": "stdClass",
                        "iss": "<redacted>/application/o/nextcloud/",
                        "sub": "<redacted>",
                        "aud": "<redacted>",
                        "exp": 1735375025,
                        "iat": 1735374725,
                        "auth_time": 1735374621,
                        "acr": "goauthentik.io/providers/oauth2/default",
                        "amr": [
                            "pwd"
                        ],
                        "nonce": "<redacted>",
                        "sid": "<redacted>",
                        "email": "<redacted>",
                        "email_verified": true,
                        "name": "<redacted>",
                        "groups": [
                            "<redacted>"
                        ],
                        "quota": "<redacted>",
                        "user_id": "<redacted>",
                        "role": "<redacted>",
                        "organisation": "<redacted>",
                        "website": "<redacted>",
                        "street_address": "<redacted>",
                        "postal_code": "<redacted>",
                        "region": "<redacted>",
                        "country": "<redacted>"
                    },
                    {
                        "__class__": "OC\\User\\User"
                    }
                ]
            },
            {
                "file": "/var/www/html/lib/private/AppFramework/Http/Dispatcher.php",
                "line": 208,
                "function": "code",
                "class": "OCA\\UserOIDC\\Controller\\LoginController",
                "type": "->",
                "args": [
                    "NAFPYW3A0TJC8B5OCPVRA27WS97XJS46",
                    "99e351ca29524ce2acffb040168c66ab",
                    "",
                    "",
                    ""
                ]
            },
            {
                "file": "/var/www/html/lib/private/AppFramework/Http/Dispatcher.php",
                "line": 114,
                "function": "executeController",
                "class": "OC\\AppFramework\\Http\\Dispatcher",
                "type": "->",
                "args": [
                    {
                        "__class__": "OCA\\UserOIDC\\Controller\\LoginController"
                    },
                    "code"
                ]
            },
            {
                "file": "/var/www/html/lib/private/AppFramework/App.php",
                "line": 161,
                "function": "dispatch",
                "class": "OC\\AppFramework\\Http\\Dispatcher",
                "type": "->",
                "args": [
                    {
                        "__class__": "OCA\\UserOIDC\\Controller\\LoginController"
                    },
                    "code"
                ]
            },
            {
                "file": "/var/www/html/lib/private/Route/Router.php",
                "line": 302,
                "function": "main",
                "class": "OC\\AppFramework\\App",
                "type": "::",
                "args": [
                    "OCA\\UserOIDC\\Controller\\LoginController",
                    "code",
                    {
                        "__class__": "OC\\AppFramework\\DependencyInjection\\DIContainer"
                    },
                    {
                        "_route": "user_oidc.login.code"
                    }
                ]
            },
            {
                "file": "/var/www/html/lib/base.php",
                "line": 1003,
                "function": "match",
                "class": "OC\\Route\\Router",
                "type": "->",
                "args": [
                    "/apps/user_oidc/code"
                ]
            },
            {
                "file": "/var/www/html/index.php",
                "line": 24,
                "function": "handleRequest",
                "class": "OC",
                "type": "::",
                "args": []
            }
        ],
        "File": "/var/www/html/lib/private/Accounts/AccountProperty.php",
        "Line": 65,
        "message": "Invalid scope",
        "exception": [],
        "CustomMessage": "Invalid scope"
    },
    "id": "676fb7e20c96f"
}
```
or am I doint something wrong?
